### ISSUE TITLE
issue with proxy with username and password

### ISSUE THREAD

**Issue by edwardchenchen** [twintproject/twint#694]
I have the code 
according to https://github.com/twintproject/twint/issues/570#issuecomment-549515548

        c.Proxy_host = 'user:password@au210.nordvpn.com'
        c.Proxy_port = 1080
        c.Proxy_type = 'socks5'
### Description of Issue
i can use a proxy without auth, but when using a proxy requires auth,
socket.gaierror: [Errno 8] nodename nor servname provided, or not known
i think it failed when doing dns request
### Full Trace

  File "/Users/echen/anaconda3/envs/cc37/lib/python3.7/site-packages/twint/run.py", line 194, in run
    await self.tweets()
  File "/Users/echen/anaconda3/envs/cc37/lib/python3.7/site-packages/twint/run.py", line 132, in tweets
    await self.Feed()
  File "/Users/echen/anaconda3/envs/cc37/lib/python3.7/site-packages/twint/run.py", line 49, in Feed
    response = await get.RequestUrl(self.config, self.init, headers=[("User-Agent", self.user_agent)])
  File "/Users/echen/anaconda3/envs/cc37/lib/python3.7/site-packages/twint/get.py", line 119, in RequestUrl
    response = await Request(_url, params=params, connector=_connector, headers=headers)
  File "/Users/echen/anaconda3/envs/cc37/lib/python3.7/site-packages/twint/get.py", line 143, in Request
    return await Response(session, url, params)
  File "/Users/echen/anaconda3/envs/cc37/lib/python3.7/site-packages/twint/get.py", line 148, in Response
    async with session.get(url, ssl=True, params=params, proxy=httpproxy) as response:
  File "/Users/echen/anaconda3/envs/cc37/lib/python3.7/site-packages/aiohttp/client.py", line 1012, in __aenter__
    self._resp = await self._coro
  File "/Users/echen/anaconda3/envs/cc37/lib/python3.7/site-packages/aiohttp/client.py", line 483, in _request
    timeout=real_timeout
  File "/Users/echen/anaconda3/envs/cc37/lib/python3.7/site-packages/aiohttp/connector.py", line 523, in connect
    proto = await self._create_connection(req, traces, timeout)
  File "/Users/echen/anaconda3/envs/cc37/lib/python3.7/site-packages/aiohttp/connector.py", line 859, in _create_connection
    req, traces, timeout)
  File "/Users/echen/anaconda3/envs/cc37/lib/python3.7/site-packages/aiohttp/connector.py", line 986, in _create_direct_connection
    req=req, client_error=client_error)
  File "/Users/echen/anaconda3/envs/cc37/lib/python3.7/site-packages/aiohttp_socks/connector.py", line 57, in _wrap_create_connection
    await proxy.connect(host, port)
  File "/Users/echen/anaconda3/envs/cc37/lib/python3.7/site-packages/aiohttp_socks/proxy/base_proxy.py", line 30, in connect
    family, host = await self._resolve_proxy_host()
  File "/Users/echen/anaconda3/envs/cc37/lib/python3.7/site-packages/aiohttp_socks/proxy/base_proxy.py", line 80, in _resolve_proxy_host
    return await self.resolve(host=host)
  File "/Users/echen/anaconda3/envs/cc37/lib/python3.7/site-packages/aiohttp_socks/proxy/mixins.py", line 50, in resolve
    family=family, type=socket.SOCK_STREAM)
  File "/Users/echen/anaconda3/envs/cc37/lib/python3.7/asyncio/base_events.py", line 788, in getaddrinfo
    None, getaddr_func, host, port, family, type, proto, flags)
  File "/Users/echen/anaconda3/envs/cc37/lib/python3.7/concurrent/futures/thread.py", line 57, in run
    result = self.fn(*self.args, **self.kwargs)
  File "/Users/echen/anaconda3/envs/cc37/lib/python3.7/socket.py", line 752, in getaddrinfo
    for res in _socket.getaddrinfo(host, port, family, type, proto, flags):
socket.gaierror: [Errno 8] nodename nor servname provided, or not known
### Environment Details
>Using Windows, Linux? What OS version? Running this in Anaconda? Jupyter Notebook? Terminal?

**Comment by crawlingcity** [comment]
Hey,

So, i was having the same issue and when I tried to authenticate via URL (basic auth), I had the following error:
`aiohttp.client_exceptions.InvalidURL: http://username:password@host:port`

I modified twint/get.tpy to receive proxyauth as  an argument, making it possible to use another param like this:

Basic idea, maintaining the current configuration would be

get.py line 67
`proxyauth = aiohttp.BasicAuth(config.Proxy_Username, config.Proxy_Password)`
get.py line 150
`async with session.get(url, ssl=True, params=params, proxy=httpproxy, proxy_auth=proxyauth) as response:`

And ofc, adding the new config params to config.py

I understand there could be another way to handle this to not break the current configuration, for those that have it working with the basic auth in the url, but let's face it, being able to make call like this, sounds like a proper way to do things:

```
        c.Proxy_host = os.getenv('PROXY_HOST')
        c.Proxy_port = os.getenv('PROXY_PORT')
        c.Proxy_username = os.getenv('PROXY_USERNAME')
        c.Proxy_password = os.getenv('PROXY_PASSWORD')
        c.Proxy_type = 'http'
```

I'll be more than willing to make a PR, if @pielco11 agrees with this approach

**Comment by pielco11** [comment]
`Proxy_username` and `Proxy_password` are not available, you should still be available to use the proxy with the "usual" format: `type://user:pass@host:port`

**Comment by crawlingcity** [comment]
I know, I can make them available, if I get some free time on my hands (already did on my project and it's working good, but need to see the big picture to not break other stuff)

Unfortunately, with the service I was using (smartproxy) aiohttp just throws an error like I said in the previous post: aiohttp.client_exceptions.InvalidURL

So, not blaming twint on this one, it's obviously aiohttp not handling it correctly, but, looking into the issue, I noticed there was more flags I could pass.

Tldr: I'll try and patch it this week

**Comment by G-Pavel** [comment]
> 
> 
> I know, I can make them available, if I get some free time on my hands (already did on my project and it's working good, but need to see the big picture to not break other stuff)
> 
> Unfortunately, with the service I was using (smartproxy) aiohttp just throws an error like I said in the previous post: aiohttp.client_exceptions.InvalidURL
> 
> So, not blaming twint on this one, it's obviously aiohttp not handling it correctly, but, looking into the issue, I noticed there was more flags I could pass.
> 
> Tldr: I'll try and patch it this week

Hello, is there any progress on this issue?

Code like this:

```
c.Proxy_host = 'user: password@au210.nordvpn.com'
c.Proxy_port = 1080
c.Proxy_type = 'socks5'
```

Still does not work ((

**Comment by crawlingcity** [comment]
I left my previous company a month after making my original comment and stopped using this and no longer have access to the projects (this explains why no PR was done).

Still, just read my previous [comment](https://github.com/twintproject/twint/issues/694#issuecomment-613470519) in which I explain the workaround I used.

Best of luck

**Comment by G-Pavel** [comment]
> 
> 
> I left my previous company a month after making my original comment and stopped using this and no longer have access to the projects (this explains why no PR was done).
> 
> Still, just read my previous [comment](https://github.com/twintproject/twint/issues/694#issuecomment-613470519) in which I explain the workaround I used.
> 
> Best of luck

OK, thanks for answer.
I found this place
```
Async with Session.get (URL, SSL = True, Params = Params, Proxy = httpproxy, proxy_auth = proxyauth) AS Response:
Get.py.
```
But makes difficult to search "Line 67" where need to insert
` Proxyauth = Aiohttp.basicauth (config.proxy_username, config.proxy_password)`
Could you tell me a new line for the current file?
